File: C++/programas/POO-Abstraccion/POO-Abstraccion/Esfera.cpp
#define _USE_MATH_DEFINES //Define el uso de constantes matemáticas.
#include "Esfera.h" //Cabecera de clase Esfera.
#include <math.h> //Cabecera para trabajar con funciones matemáticas.
#include <iostream> //Cabecera para entrada y salida de datos.

//Inicialización de atributos por medio del constructor.
Esfera::Esfera(double radio) { //Una forma de inicializar el constructor.
	Radio = radio;
}

//Declaración de métodos abstractos en clase derivada. Regresa volumen de una esfera.
double Esfera::CalcularVolumen() {
	return (4 * M_PI * pow(Radio, 3)) / 3;
}

//Declaración de métodos abstractos en clase derivada. Despliefa datos de una esfera.
void Esfera::DesplegarVolumen() {
	std::cout << "\n--Datos del esfera--\n\n";
	std::cout << "La medida del radio de la esfera es: " << Radio << "u.\n";
	std::cout << "El volumen de la esfera es: " << CalcularVolumen() << "u^3.\n";
}

File: Java/programas/PO-Abstracción/src/Esfera.java
//Declara clase regular derivada de Figuras.
public class Esfera extends Figuras {
    private double Radio; //Atributo privado.

    public Esfera(double radio){ //Constructor que inicializa atributo.
        this.Radio = radio;
    }

    public double CalcularVolumen(){ //Función que regresa el volumen de la Esfera.
        return (4 * Math.PI * Math.pow(Radio, 3)) / 3;
    }

    public void DesplegarVolumen(){ //Procedimiento que despliega el radio y volumen de la esfera.
        System.out.println("\n---Datos de la esfera---\n");
        System.out.println("La medida del radio de la esfera es:" + Radio + "u.");
        System.out.println("El volumen de la esfera es: " + CalcularVolumen() + "u^3.");
    }
}